hash-table/src/hash_table.c

```diff
@@ -9,8 +9,7 @@
 
 
 // HT_DELETED_ITEM is used to mark a bucket containing a deleted item
-static ht_item* HT_DELETED_ITEM = &(ht_item) {NULL, NULL};
```

Comment:
I don't know about the `&(ht_item)` syntax; never seen that before.  For a global sentinel like this, declare a non-pointer, and either use `&HT_DELETED_ITEM` (preferred), or make a separate pointer to point to the value object.